### ISSUE TITLE
[Feature Request] Group Transaction

### ISSUE THREAD

**Issue by 8227846265** [jameskokoska/Cashew#216]
## Is your feature request related to a problem? Please describe.
Hello, could you add a feature for `grouping transactions`?

For instance, consider a situation where I make a purchase at Walmart. This purchase spans several expense categories and sub-categories but is covered by a single transaction, paid for with a credit card.

Details of the Single Transaction:
- Total Receipt Amount: $500 (Paid by Credit Card)

This single transaction includes items from two main categories, each with its own sub-categories:
A. Groceries
- Meat & Seafood
- Canned Goods

B. Shopping
- Kitchen Supplies
- Bathroom Supplies

##  Transaction Receipt Breakdown Example in Cashew via Group Transaction

DATE: April 8, 2024, 11 AM
ACCOUNT: Credit Card
AMOUNT: $500
TEXT TITLE: Walmart
TEXT DESCRIPTION: Bought weekly supplies for home.

BREAKDOWN:
A. Groceries: $320 Total

  Meat & Seafood: $180
    - Fresh salmon fillet, 2 lbs: $40
    - Ribeye steak, 2 lbs: $50
    - Chicken breasts, 3 lbs: $30
    - Shrimp, 2 lbs: $60
    
  Canned Goods: $140
    - Diced tomatoes, 4 cans: $8
    - Black beans, 4 cans: $8
    - Corn, 4 cans: $8
    - Chicken broth, 4 cartons: $12
    - Tuna, 6 cans: $15
    - Olive oil, 1 liter: $15
    - Coffee, 1 lb: $15
    - Pasta, 5 packs: $10
    - Tomato sauce, 4 jars: $20
    - Peanut butter, 2 jars: $10
    - Almonds, 1 lb: $19

B. Shopping: $180 Total

  Kitchen Supplies: $110
    - Knife set: $30
    - Cutting board: $20
    - Mixing bowls, set of 3: $25
    - Measuring cups and spoons: $15
    - Spatula set: $10
    - Tea towels, set of 4: $10
    
  Bathroom Supplies: $70
    - Bath towels, set of 4: $30
    - Toiletries (shampoo, soap, toothpaste): $20
    - Toilet paper, 12 rolls: $10
    - Bathroom cleaner, 2 bottles: $10

## Why group transactions?
- The credit card statement will only show a $500 expense at Walmart. 
- Using Cashew to group transactions can help accurately record the $500 spent and assign the correct expense categories for each item purchased without needing to create dozens of separate records.

## UI/UX: How to create a group transaction
1. Click +
2. Ask user whether transaction is single or group transaction.
3. If group transaction, user is able to create multiple records with different categories and sub-categories
4. Once the recording is done. The transaction will appear as one record on Transaction History but once clicked. It will show the breakdown of the transaction in amount/categories/sub-categories.

**Comment by jameskokoska** [comment]
Cashew was always meant for a quick transaction log, therefore in this example a 500$ single transaction makes the most sense. Cashew was not intended and designed in a way for users to regurgitate every single detail of every purchase. However, everyone budgets differently and Cashew still allows users to have the freedom to log down their spending in any way they prefer. You can create separate transactions for as many detailed line items within a bill you wish. Adding a transaction in Cashew is very fluid, efficient, and quick and allows for this use case already.

In short: to achieve 'grouped transactions' the user will need to add a separate transaction for every purchase detail they see fit. This gives users the freedom to choose the detail level of logging while keeping the interface straightforward and simple.

**Comment by ngocanhtve** [comment]
I think this would be a cool and unique feature if implemented. However, I also feel that it will certainly be a major technical barrier and will create more complexity in terms of functionality and user interface.

**Comment by 8227846265** [comment]
@jameskokoska, is there any way to create a link between multiple transaction records for a single receipt?

Often, when reviewing my credit card expenses from last month and recording them as separate records, I always wonder where this $500 is coming from.

For example, my Cashew transaction was split into 3 records: $200, $100, and $100.

I'm looking for some sort of signal that indicates these 3 records are part of the same transaction but split.

**Comment by jameskokoska** [comment]
> However, I also feel that it will certainly be a major technical barrier and will create more complexity in terms of functionality and user interface.

I agree. Cashew always prioritizes ease of use and is a beginner budgeting tool.  If one wants to budget with this level of detail, Cashew does allow for this with individual transactions. However, if one also wants this level of detail, another budgeting application/solution may be a better choice (such as a spreadsheet)

**Comment by 8227846265** [comment]
I get it that we can fluidly create records but linking records into a single receipt transaction and making sense of split transactions is very hard to do now in Cashew.

**Comment by jameskokoska** [comment]
> is there any way to create a link between multiple transaction records for a single receipt?

I'm assuming since you log them for the same time period, they will be already chronologically grouped together and possibly have the same title.

If you want to quickly view the total, you can swipe to select the transactions. Similarly, the date banner total will show you the total amount spent on that day.

**Comment by 8227846265** [comment]
> If you want to quickly view the total, you can swipe to select the transactions. Similarly, the date banner total will show you the total amount spent on that day.

This may not be feasible if multiple single receipt transactions occur in a day.

I'll think about it more later and post here possible solutions that is easy to implement without changing too much on the current UI/UX of adding new transactions and transaction history module of the app.
